x-pack/plugins/actions/server/lib/get_token_count_from_openai_stream.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { encode } from 'gpt-tokenizer';
```

Comment:
do we need to consider the encoding model? Since we are allowing free text for the model, the user could be using a model that does not support `cl100k_base` encodings

Reply:
The API we use (`/v1/chat/completions`) is gpt-3.5-turbo and gpt-4 only: https://platform.openai.com/docs/models/model-endpoint-compatibility

Reply:
Ah. We should update the field description on the create connector form to specify that